test/e2e/framework/framework.go

```diff
@@ -262,7 +262,7 @@ func (f *Framework) matchNginxConditions(name string, matcher func(cfg string) b
 				glog.Infof("nginx.conf:\n%v", o)
 			}
 
-			if matcher(o) {
+			if matcher(strings.Join(strings.Fields(o), " ")) {
```

Comment:
Add this there are sections with some variable spaces within the template

strings.Fields(o) = splits everything up by 1 or more spaces, then we join each item, with spaces in-between